tests/phpunit/integration/Core/Modules/ModulesTest.php

```diff
@@ -268,10 +268,12 @@ public function test_is_module_connected() {
 		$valid_module_slug = 'search-console';
 		$this->assertArrayHasKey( $valid_module_slug, $modules->get_available_modules() );
 		$this->assertTrue( $modules->is_module_connected( $valid_module_slug ) );
+		$this->assertTrue( apply_filters( 'googlesitekit_is_module_connected', false, $valid_module_slug ) );
```

Comment:
The filter here isn't really testing what we expect because `$modules` here isn't the same instance that is handling the callback (since this test doesn't call `$modules->register()`) – it uses the main instance created during plugin bootstrap. That doesn't really change the result in this case, but we still shouldn't do it.

We can simplify the approach here a bit to be consistent with the test on `googlesitekit_module_exists` in `test_register`. We should update it to first clear all the existing filters for those which we make assertions about which isn't currently done there. Some test cases (with too much in `register`) extract the specific filter tests to dedicated named variants (e.g. `register__googlesitekit_is_module_connected`) or something like that to avoid testing too many things in a single test, which isn't probably necessary here but would be good if we were to go into more thorough testing of the behavior.

Reply:
This one is yet to be addressed as well.